IntelliTect.TestTools.Selenate/IntelliTect.TestTools.Selenate/ConditionalWait.cs

```diff
@@ -0,0 +1,113 @@
+﻿using OpenQA.Selenium;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IntelliTect.TestTools.Selenate
+{
+    public class ConditionalWait
```

Comment:
I will leave this up to you to decide, but looking at the method signatures it was not immediately apparent to me what each of the generic parameters represented. 
IMO I would rather work with an API more like this:
`await ConditionalWait.WaitFor(() => GetThingy()).Ignore<Exception1, Exception2>();`
Thoughts? If you like it and want help putting something like this together let me know.

Reply:
I struggled to come up with a good solution to this. I'll skip it for now, but would like to get together with you to chat about it more. TL;DR, I had trouble getting the return type from WaitFor to a final Wait() method.

Reply:
ah yes happy to sit down with you sometime can go over how to structure the code.